service/service.go

```diff
@@ -107,8 +109,18 @@ func New(tarball io.Reader, options ...Option) (*Service, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	s.injectDefinition(def)
 
+	if err := s.validateConfigurationEnv(env); err != nil {
+		return nil, err
+	}
+
+	// if the all keys exists append to configuration env.
+	// The order of variables allows to safely append new variables
+	// without removing previous one. The last variable in slice will take precedens.
+	s.configuration.Env = append(s.configuration.Env, xos.EnvMapToSlice(env)...)
```

Comment:
I don't get the behaviour here:

I'm assuming that we have two rules here:
* Env vars can be defined with **defaults** inside the mesg.yml.
* Some of **these(pre-defined)** env variables can be overwritten by the deploy command during service's deployment.

But the code here appends env variables provided from deploy command to the ones defined inside the yml. But it needs to overwrite the defaults instead.

Reply:
missing tests for new behaviour

Reply:
> But the code here appends env variables provided from deploy command to the ones defined inside the yml. But it needs to overwrite the defaults instead.

So this is the comment about. It doesn't have to overwrite, because of env works. If you have two in a slice, the one that is lather will take the precedence.

Reply:
correct. but there is no need to save duplicated env variables. they should be removed. otherwise, they'll use extra space in the storage and we'll see this unused env variables everywhere and in the service detail command which doesn't make sense.

Reply:
> we'll see this unused env variables everywhere and in the service detail command which doesn't make sense.

You don't store the configuration because it's not exported.

Reply:
https://github.com/mesg-foundation/core/blob/62934d5d94f418669f2a960480ca143f11703921/service/service.go#L219

Reply:
good catch